Subsets fixes; and Scaling conformance class

jerstlouis opened the pull request.

jerstlouis commented:
Note that I have gone ahead and made the changes regarding the subset requirement (in a separate commit: https://github.com/opengeospatial/ogc_api_coverages/pull/88/commits/508f268067c841193ab3e2ed37a22e8dd1a96f6b), because it is obvious from the existing examples in the current specifications
(https://github.com/opengeospatial/ogc_api_coverages/blob/master/standard/examples/examples_subsetting.adoc) that `subset` was intended to work on the rangeset resource.

Schpidi commented on standard/examples/examples_subsetting.adoc:
```diff
@@ -1,6 +1,6 @@
-*   http://acme.com/oapi/collections/%7Bcoverageid%7D/coverage?subset=Lat(40,50)&subset=Long(10,20)["http://acme.com/oapi/collections/{coverageid}/coverage?subset=Lat((40,50)&subset=Long(10,20)"] -- returns a cutout from the coverage identified with extent between corner coordinates (40,10) and (50,20) (note: replace the _ in the URL by a comma character - an adoc issue)
-*   http://acme.com/oapi/collections/%7Bcoverageid%7D/coverage/rangeset?subset=Lat(40,50)&subset=Long(10,20)["http://acme.com/oapi/collections/{coverageid}/coverage/rangeset?subset=Lat(40,50)&SUBSET=Long(10,20)"]  -- returns the range set of a coverage cutout between (40,10) and (50,20), in the coverage’s Native Format; no domain set, range type, and metadata will be returned (note: replace the _ ...
-*   http://acme.com/oapi/collections/%7Bcoverageid%7D/coverage?subset=time(%222019-03-27%22)["http://acme.com/oapi/collections/{coverageid}/coverage?subset=time(\"2019-03-27\")"] -- returns a coverage slice at the timestamp given (in case the coverage is Lat/Long/time the result will be a 2D image)
+* {datasetAPI}/collections/{coverageid}/coverage/rangeset?subset=Lat(40:50),Lon(10:20) -- returns a coverage cutout (sample values only) between (Lat: 40, Lon: 10) and (Lat: 50, Lon: 20), in the negotiated format
```
Could be deleted as it is the same example as the last one, isn't it?

jerstlouis replied:
Is it about `/coverage/rangeset` vs. `/coverage` ?

Schpidi commented on standard/requirements/coverage-scaling/REQ_cov-scaling-definition.adoc:
```diff
@@ -0,0 +1,31 @@
+  axisName:          {NCName}
+
+  Where:
+     {NCName} is an XML-style identifier not containing ":" (colon) characters,
```
Why limit to NCName and why mention XML?

jerstlouis replied:
I copied this from the subset requirements which mentioned this.
That requirement not to have a colon might come from WCS?

pebau replied:
> I copied this from the subset requirements which mentioned this.
> That requirement not to have a colon in the axis name might come from WCS?

Indeed, this is relevant specifically for XML encodings, dating back to before coverages emancipated from GML. In CIS 1.1 such restrictions consequently hold when using the particular encodings, so in this case would constrain only XML-encoded (maybe also RDF?) requests. Wouldn't make sense for JSON indeed.

Schpidi commented:
Coverages SWG call: Agreed to merge and apply the following fixes afterwards
- Remove duplicate subset example
- Remove mention of NCName and XML
- Adjust to using EBNF instead of BNF

jerstlouis commented:
Addressed issues in https://github.com/opengeospatial/ogc_api_coverages/pull/98
(EBNF might need more informed tweaks / thorough review still though)
